docs/topics/serialization.md

```diff
@@ -0,0 +1,78 @@
+[//]: # (title: Serialization)
+
+**Serialization** is the process of converting data used by an application to a format that can be transferred over a
+network or stored in a database or a file. Deserialization is the opposite process of converting external data back into a runtime object.
+Together, they are essential to most applications that exchange data with third parties.
+
+Some data serialization formats, such as [JSON](https://www.json.org/json-en.html) and [Protocol Buffers](https://protobuf.dev/) are particularly common.
+Being language-neutral and platform-neutral, these formats enable data exchange between systems written in any modern language.
+
+To convert an object tree to a string or to a sequence of bytes, it must go through two mutually intertwined processes: 
+
+1. Serialization: Objects are transformed into a sequence of their primitive values.
+This universal process varies depending on the object and is managed by a serializer.
+2. Encoding: The primitive sequence is converted into the desired output format, controlled by an encoder.
+
+![Serialization flow](serialization.svg){width=700}
+
+The reverse process involves parsing the input format, decoding the primitive values, and then deserializing the resulting
+stream into objects.
```

Comment:
The reverse process involves the decoder invoking the input format to guide it in parsing the input (there is no intermediate format and that cannot exist - note that formats may not allow parsing without the knowledge of specific data structures (generally non-self-describing binary formats)).

Reply:
Maybe an interesting observation to add somewhere is that while encoding can generally simply skip elements either in the serializer or in the format, this is not true for decoders. Decoders must read the byte/text streams and consume them (even if it is just to explicitly ignore the parsing results).